lms/djangoapps/verify_student/views.py

```diff
@@ -256,19 +256,36 @@ def get(
             log.warn(u"No course specified for verification flow request.")
             raise Http404
 
-        # Verify that the course has a verified mode
-        course_mode = CourseMode.verified_mode_for_course(course_key)
+        # Check that the course has an unexpired verified mode
+        course_mode, expired_course_mode = self._get_verified_modes_for_course(course_key)
+
         if course_mode is None:
```

Comment:
Nitpick, but you may be able to simplify 262-288 to be

if course_mode is not None:
    ...
elif expired_course_mode is not None:
    ...
else:
    ...